homeassistant/components/reolink/light.py

```diff
@@ -108,8 +108,7 @@ def is_on(self) -> bool:
     @property
     def brightness(self) -> int | None:
         """Return the brightness of this light between 0.255."""
-        if self.entity_description.get_brightness_fn is None:
-            return None
```

Comment:
If it's not optional, why don't we make the function required?

Reply:
The function is optional, but if the function is not defined (defaults to None),
You get:
`self._attr_supported_color_modes = {ColorMode.ONOFF}`
while otherwise you have
`self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}`

So when the function is set to None, the entity is set to not support BRIGHTNESS and the `brightness` property can never be called.
Therefore it is inpossible to cover that line in the tests, because that line will never be called.

That is why I made a assert in there.